keygen/src/keygen.rs

```diff
@@ -726,22 +728,25 @@ fn do_main(matches: &ArgMatches<'_>) -> Result<(), Box<dyn error::Error>> {
                                 grind_matches_thread_safe[i]
                                     .count
                                     .fetch_sub(1, Ordering::Relaxed);
-                                if !no_outfile {
-                                    write_keypair_file(&keypair, &format!("{}.json", keypair.pubkey()))
-                                    .unwrap();
-                                    println!(
-                                        "Wrote keypair to {}",
-                                        &format!("{}.json", keypair.pubkey())
-                                    );
-                                }
+
+                                let outfile = {
+                                    if no_outfile {
+                                        STDOUT_OUTFILE_TOKEN.to_string()
+                                    } else {
+                                        format!("{}.json", keypair.pubkey())
+                                    }
+                                };
+                                let _ = match output_keypair(&keypair, &outfile, "grind") {
+                                    Err(e) => println!("Unable to write {}: {}", outfile, e),
+                                    _ => {}
+                                };
```

Comment:
this is a little less verbose:
```suggestion
                                output_keypair(&keypair, &outfile, "grind")
                                    .unwrap_or_else(|e| eprintln!("Unable to write {}: {}", outfile, e));
```

Reply:
Good call on both of these, thanks!